pywhy_stats/categorical.py

```diff
@@ -0,0 +1,293 @@
+
+import numpy as np
+from numpy.typing import ArrayLike
+from scipy import stats
+from sklearn.preprocessing import LabelEncoder
+
+from .pvalue_result import PValueResult
+
+
+def ind(X: ArrayLike, Y: ArrayLike, lambda_: str = "cressie-read") -> PValueResult:
+    """Perform an independence test using power divergence test.
+
+    Parameters
+    ----------
+    X : ArrayLike of shape (n_samples,)
+        The first node variable.
+    Y : ArrayLike of shape (n_samples,)
+        The second node variable.
+    lambda_ : float or string
+        The lambda parameter for the power_divergence statistic. Some values of
+        ``lambda_`` results in other well known tests:
+            "pearson"             1          "Chi-squared test"
+            "log-likelihood"      0          "G-test or log-likelihood"
+            "freeman-tukey"     -1/2        "freeman-tukey Statistic"
+            "mod-log-likelihood"  -1         "Modified Log-likelihood"
+            "neyman"              -2         "Neyman's statistic"
+            "cressie-read"        2/3        "The value recommended in the paper
+                                             :footcite:`cressieread1984`"
+
+    Returns
+    -------
+    statistic : float
+        The test statistic.
+    pvalue : float
+        The p-value of the test.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    return _power_divergence(X=X, Y=Y, Z=None, lambda_=lambda_)
+
+
+def condind(
+    X: ArrayLike, Y: ArrayLike, condition_on: ArrayLike, lambda_: str = "cressie-read"
+) -> PValueResult:
+    """Perform an independence test using power divergence test.
+
+    Parameters
+    ----------
+    X : ArrayLike of shape (n_samples,)
+        The first node variable.
+    Y : ArrayLike of shape (n_samples,)
+        The second node variable.
+    condition_on : ArrayLike of shape (n_samples, n_variables)
+        The conditioning set.
+    lambda_: float or string
+        The lambda parameter for the power_divergence statistic. Some values of
+        lambda_ results in other well known tests:
+            "pearson"             1          "Chi-squared test"
+            "log-likelihood"      0          "G-test or log-likelihood"
+            "freeman-tukey"     -1/2        "freeman-tukey Statistic"
+            "mod-log-likelihood"  -1         "Modified Log-likelihood"
+            "neyman"              -2         "Neyman's statistic"
+            "cressie-read"        2/3        "The value recommended in the paper
+                                             :footcite:`cressieread1984`"
+
+    Returns
+    -------
+    statistic : float
+        The test statistic.
+    pvalue : float
+        The p-value of the test.
+    """
+    return _power_divergence(X=X, Y=Y, Z=condition_on, lambda_=lambda_)
+
+
+def _compute_conditional_contingency_table(X, Y, Z):
+    unique_Z_vals = np.unique(Z)
+
+    chi = 0
+    dof = 0
+
+    for z_state in unique_Z_vals:
+        pass
```

Comment:
This seems to be a left-over template and is overridden by the next method definition in line 103. Just remove.